app/src/main/java/com/dontstopthemusic/dontstopthemusic/FirstFragment.java

```diff
@@ -102,12 +110,31 @@ public void onClick(View view) {
                         break;
                     }
                     case TWO_HitPFL:{
-                        if (!(TEST_monitorFeedback.length==0)){
+                        if (!(TEST_monitorFeedback.length()==0)){
                             //feedback exists in this channel
                             current_state=FeedbackStates.THREE_CheckFreqRange;
                             //stuff happens to determine the problem area
-                            String problem_area="HIGH";
-                            tv1.setText("We have found a potential problem channel. The feedback is centered around the "+problem_area+" range. Try turning down the "+problem_area+" knob.");
+                            int average=0; //brute ugly average
+                            for (int i=0;i<TEST_monitorFeedback.length();i++){
+                                try {
+                                    average=average+TEST_monitorFeedback.getInt(i);
+                                } catch (JSONException e) {
+                                    throw new RuntimeException(e);
+                                }
```

Comment:
Just rethrow by doing `throw e` instead of making new exception and losing stack trace